### PR TITLE
Update circle CI to allow for trunk based feature branch development

### PR DESCRIPTION
**Pull request by adaviesMOJ**
## What does this change do?

Rework main flow to be exclusive to main, with automatic deployment up to staging.

Introduce feature flow which requires approval and terminates at staging.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -41,35 +41,6 @@ jobs:
...
-      - run:
-          name: Change to project directory then clean install
-          command: |
-            cd hmpps-vsip-ui-tests
-            mvn -f pom.xml clean install -Dmaven.test.skip=true
-      - run:
-          name: Change to project directory and run tests
-          command: |
-            cd hmpps-vsip-ui-tests
-            mvn test -Dbrowser=remote-chrome -Dspring.profiles.active="staging" -Dtest=uk.gov.justice.digital.hmpps.vsip.suites.RunSuite
```
Comment (dhirajshettymoj) [inline .circleci/config.yml:72]: are we not doing these tests on orchestration service now?

File: .circleci/config.yml [hunk 2]
```diff
@@ -165,73 +153,74 @@ workflows:
...
+  build-test-deploy-feature:
     jobs:
       - validate:
           filters:
             tags:
               ignore: /.*/
+            branches:
+              ignore:
+                - main
+
```
Comment (dhirajshettymoj) [inline .circleci/config.yml:176]: should we look for a branch name with feature in it?

Comment (dhirajshettymoj) [reply .circleci/config.yml:176]: comment from @adaviesMOJ on slack - it doesn't need the feature- naming convention, as this flow can now be reused for security patches etc. So we will have 2 flows "main" and "other"

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Review by AledWEvans — APPROVED** [review]
LGTM

**Pull request merged**